docs/contributing/999-rfcs/2022-12-08-wing-inputs.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "#2726 Wing Inputs"
+description: "How to create, share, manage inputs"
+---
+
+# Wing Inputs
+- **Author(s)**: @Hasanaburayyan
+- **Submission Date**: {2023-12-08}
+- **Stage**: Draft
+
+Using Wing inputs in your application
+
+## Background
+
+Frequently applications have the need to read inputs, whether those inputs are sensitive data like passwords, connection strings, ssh keys or non-sensitive configuration such as retry counts, endpoint urls, etc..
+
+## Use cases
```

Comment:
This is a great list. Here are some more user stories to consider (there might be overlap with what you wrote):

- As a user, I would like to write a cloud application, and have a secret loaded from a local file (like `.env` or `~/.wing/secrets` etc.) during local testing.

- As a user, I would like to write a cloud application, and then deploy it in two different AWS/Azure/Cloudflare/etc accounts by applying a different set of input values to each deployment.

- As a user, I would my cloud application to automatically meet compliance requirements by automatically rotating application secrets every 90 days.

- As a user, I would like my Lambda function to securely use a database connection string that is the [sensitive](https://developer.hashicorp.com/terraform/language/functions/sensitive) output of a third-party terraform resource. (This is a rephrasing of the use case for the `postgres` library where Neon's `Database` resource exposes a connection string as a CDKTF token, and we want that token to be safely resolved and loaded by inflight application code)

- As a user, I would like to write a cloud application, and have different staging environments have different scaling parameters according to an input I provide at deploy time. These parameters are not sensitive values, and are safe to emit to logs or non-sensitive infrastructure configuration like environment variables.

An interesting thought experiment could be to decompose all of the use cases into three constituent lists -- we essentially have:

1. The kinds of values inputs take on (numbers, URLs, sensitive strings, structured objects, etc.)
2. The ways inputs are provided or generated (they could be entered by user, loaded from disk, generated ad-hoc preflight, generated ad-hoc inflight, read from a third party source, resolved at deployment time, etc.)
2. The ways inputs are consumed (they could be used to generate different infrastructure in preflight code e.g. `for i in input.asNum() { ... }`, they could be used inside inflight code, etc.)

Reply:
There is another type of inputs which are configuration values that are needed by the target platform provider.

For example, specifying if container workloads are deployed to EKS or ECS (and the respective configuration thereof).

Another example is route53 domain information and certificate ARNs which are required if a user defines a cloud.Domain resource and uses the tf-aws target.